### DISCUSSION TITLE
Scalar function: CharIndex

### DISCUSSION THREAD

**Discussion by gc-dw** [duckdb/duckdb#2562] [Q&A] [answered]
Does the duckdb support this scalar function(**CharIndex**)?

```
java.sql.SQLException: Catalog Error: Scalar Function with name charindex does not exist!
Did you mean "variance"?
	at org.duckdb.DuckDBNative.duckdb_jdbc_prepare(Native Method) ~[duckdb_jdbc-0.3.0.jar:?]
	at org.duckdb.DuckDBPreparedStatement.prepare(DuckDBPreparedStatement.java:73) ~[duckdb_jdbc-0.3.0.jar:?]
	at org.duckdb.DuckDBPreparedStatement.<init>(DuckDBPreparedStatement.java:55) ~[duckdb_jdbc-0.3.0.jar:?]
	at org.duckdb.DuckDBConnection.prepareStatement(DuckDBConnection.java:55) ~[duckdb_jdbc-0.3.0.jar:?]
	at org.duckdb.DuckDBConnection.prepareStatement(DuckDBConnection.java:168) ~[duckdb_jdbc-0.3.0.jar:?]
```

**Comment by Alex-Monahan** [comment, marked as answer]
The DuckDB equivalent is INSTR ! Just be aware that it returns a bigint, so you may need INSTR(column, 'string to find')::int

**Reply by killerfurbel** [reply]
INSTR is actually no **equivalent**, since CHARINDEX allows a third parameter (start_location): https://learn.microsoft.com/en-us/sql/t-sql/functions/charindex-transact-sql

I was just reading this blog article: https://duckdb.org/2022/05/04/friendlier-sql.html

> If you are migrating your workloads to DuckDB and a different function name would be helpful, please reach out - they are very easy to add as long as the behavior is the same!

Wouldn't it be possible to add the third parameter as well as an alias for INSTR so that it is accessible via CHARINDEX as well?

Three other functions which just need an 1:1 alias with same parameters are:
- [COUNT_BIG](https://learn.microsoft.com/en-us/sql/t-sql/functions/count-big-transact-sql) as alias for COUNT (--> COUNT_BIG is bigint, count_star in DuckDB is already int64)
- [LEN](https://learn.microsoft.com/en-us/sql/t-sql/functions/len-transact-sql) as alias for LENGTH
- [DATALENGTH](https://learn.microsoft.com/en-us/sql/t-sql/functions/datalength-transact-sql) as alias for STRLEN

I mean it's no big deal to change the queries to the DuckDB function names, but since the blog post just said "please reach out", it would definitely save some time when migrating workloads from Microsoft SQL Server.
